apache/superset
# chore: Move all dataset files/folders to the new structure

### SUMMARY

This PR moves all dataset folders and files to `src/features` and `src/pages` as seen in the new file structure discussion: https://github.com/apache/superset/discussions/22330

Notable changes:
- What was previously [AddDataset/index](https://github.com/apache/superset/blob/master/superset-frontend/src/views/CRUD/data/dataset/AddDataset/index.tsx) is now `DatasetPage`
- The types in [AddDataset/types.tsx](https://github.com/apache/superset/blob/master/superset-frontend/src/views/CRUD/data/dataset/AddDataset/types.tsx) were combined with [the type file in the root dataset folder](https://github.com/apache/superset/blob/master/superset-frontend/src/views/CRUD/data/dataset/types.ts)
- What was previously [DatasetPanel](https://github.com/apache/superset/tree/master/superset-frontend/src/views/CRUD/data/dataset/AddDataset/DatasetPanel) is now `DatasetCreationPanel`

### TESTING INSTRUCTIONS

- Go to all dataset areas and ensure that they still show up and function properly

### ADDITIONAL INFORMATION
